tags/tags-api/api/src/java/org/sakaiproject/tags/api/Tag.java

```diff
@@ -0,0 +1,253 @@
+    }
+
+    public void setCreationDate(long  creationDate) {
+        this.creationDate = creationDate;
+    }
+
+    public String getLastModifiedBy() {
+        return lastModifiedBy;
+    }
+
+    public void setLastModifiedBy(String lastModifiedBy) {
+        this.lastModifiedBy = lastModifiedBy;
+    }
+
+    public long  getLastModificationDate() {
+        return lastModificationDate;
+    }
+
+    public void setLastModificationDate(long  lastModificationDate) {
+        this.lastModificationDate = lastModificationDate;
+    }
+
+    public String getExternalId() {
+        return externalId;
+    }
+
+    public void setExternalId(String externalId) {
+        this.externalId = externalId;
+    }
+
+    public String getAlternativeLabels() {
+        return alternativeLabels;
+    }
+
+    public void setAlternativeLabels(String alternativeLabels) {
+        this.alternativeLabels = alternativeLabels;
+    }
+
+    public Boolean getExternalCreation() {
+        return externalCreation;
+    }
+
+    public void setExternalCreation(Boolean externalCreation) {
+        this.externalCreation = externalCreation;
+    }
+
+    public long  getExternalCreationDate() {
+        return externalCreationDate;
+    }
+
+    public void setExternalCreationDate(long  externalCreationDate) {
+        this.externalCreationDate = externalCreationDate;
+    }
+
+    public Boolean getExternalUpdate() {
+        return externalUpdate;
+    }
+
+    public void setExternalUpdate(Boolean externalUpdate) {
+        this.externalUpdate= externalUpdate;
+    }
+
+    public long  getLastUpdateDateInExternalSystem() {
+        return lastUpdateDateInExternalSystem;
+    }
+
+    public void setLastUpdateDateInExternalSystem(long  lastUpdateDateInExternalSystem) {
+        this.lastUpdateDateInExternalSystem = lastUpdateDateInExternalSystem;
+    }
+
+    public String getParentId() {
+        return parentId;
+    }
+
+    public void setParentId(String parentId) {
+        this.parentId = parentId;
+    }
+
+    public String getExternalHierarchyCode() {
+        return externalHierarchyCode;
+    }
+
+    public void setExternalHierarchyCode(String externalHierarchyCode) {
+        this.externalHierarchyCode = externalHierarchyCode;
+    }
+
+    public String getExternalType() {
+        return externalType;
+    }
+
+    public void setExternalType(String externalType) {
+        this.externalType = externalType;
+    }
+
+    public String getData() {
+        return data;
+    }
+
+    public void setData(String data) {
+        this.data = data;
+    }
+
+    public String getCollectionName() {
+
+        return collectionName;
+    }
+
+    public void setCollectionName(String collectionName) {
+        this.collectionName = collectionName;
+    }
+
+    /**
+     * Check that the values we've been given make sense.
+     */
+    public Errors validate() {
+        Errors errors = new Errors();
+        //TODO Validate the tag values
```

Comment:
I marked this here when I was understanding the PASystem code in case I wanted to do some extra validations in the tags values. I've never needed to do that, but it can be an improvement in the future that's why I left it. I don't feel it needed at this moment.